examples/src/main/scala/io/finch/middleware/Main.scala

```diff
@@ -0,0 +1,58 @@
+package io.finch.middleware
+
+import cats.effect.IO
+import com.twitter.finagle.Http
+import com.twitter.finagle.http.{Response, Status}
+import com.twitter.util.{Await, Time}
+import io.finch._
+
+/**
+  * Small Finch hello world application serving endpoint protected by serious authentication
+  * where each request & response are also logged and measured.
+  *
+  * This is achieved using Kleisli-based middleware together with [[Bootstrap.compile]]
+  *
+  * Use the following curl commands to test it:
+  *
+  * {{{
+  *   curl -v -H "Authorization: secret" http://localhost:8081/hello
+  *   curl -V -H "Authorization: wrong" http://localhost:8081/hello
+  * }}}
+  */
+object Main extends App with Endpoint.Module[IO] {
+
+  val helloWorld: Endpoint[IO, String] = get("hello") {
+    Ok("Hello world")
+  }
+
+  def auth(compiled: Endpoint.Compiled[IO]): Endpoint.Compiled[IO] = {
```

Comment:
I wonder if we can define these as `val`s like the examples in the [Cats Docs](https://typelevel.org/cats/datatypes/kleisli.html#configuration). I think its fine the way it is, but I might play with that later today.